### DISCUSSION TITLE
Multi tenant analytics

### DISCUSSION THREAD

**Discussion by DMaxence** [umami-software/umami#1687] [Q&A] [answered]
I have a multi tenant application where users can generate a subdomain with their data (like hashnode)
A user will then have `sub.domain.xyz`.

I want to provide for each user it's own analytics on it's subdomain.

To separate the data for more security I think it's better to create a website for each user on umami.

The way I see how to do it is:
- create a website on umami when the user creates it's website on my app
- add the user's umami website id to the page he created from my app
- then provide a link to the user where I fetch his analytics from the website id
- show the analytics from the API call done before

The problem I see in this approach is that I will have to do multiple API calls, and an insecure one
- first api call to my api to create the website 
- second api call to umami api to connect the admin account and get the Bearer token (very insecure call as the user can see my password in the payload)
- third api call to umami api to create the website with the bearer token and the infos returned by my api (url...)
- fourth api call to my api to record umami website id returned by umami third api call

And every time my user wants to see his analytics the connection to get a bearer token would need to be done again.

Is there an other way that i didn't think of to do it more securely ?
Or maybe I can integrate umami in my Next.js application to use the same bearer token ?

Thanks for your help !

**Comment by DMaxence** [comment, marked as answer]
I finally managed to do it!

For those interested I just simply did the call from my API instead of my front-end. I didn't thought of this at the beginning because my api url contained `analytics` and my ad-blocker blocked it right away.
So just renaming my API route permits me to do the call from my API...

**Reply by Prannesh45** [reply]
Hey, thanks for the response. Can you show me a demo code or repo which has this feature integrated ?

**Reply by briancao** [reply]
Umami 2.0 features teams, where you can share your website among groups of individuals without making them completely public.  If this solves your issue, I would just wait for, and use 2.0.  To be released in the coming weeks.

**Reply by Prannesh45** [reply]
Thanks Brian, my requirement is something like

Let's say I have a website like LinkedIn so we login and create a profile and we get unique landing page for user A, user B and user C. Will it be possible to get unique analytics for every landing page such that only user A can see user A analytics, user B can see user B analytics and so on. Is it feature possible/available in umami ?
